src/boot/efi/boot.c

```diff
@@ -2357,6 +2357,10 @@ static EFI_STATUS image_start(
         if (err != EFI_SUCCESS)
                 return log_error_status(err, "Error preparing initrd: %m");
 
+        /* Ask Shim to leave its protocol around, so that the stub can use it to validate PEs.
+         * By default, Shim uninstalls its protocol when calling StartImage(). */
+        shim_retain_protocol();
```

Comment:
This would be too late if the user loads drivers from `/EFI/systemd-boot/drivers`. Then again, you cannot load drivers with shim, so they'd have to be signed by `db` keys anyways, obviating the need for shim...

Reply:
good catch, moved to the beginning